docs/architecture/speculative-workflow-task.md

```diff
@@ -6,157 +6,168 @@ There are three types of Workflow Task:
   2. Transient
   3. Speculative
 
-Every Workflow Task ships history to the worker. Last events must be `WorkflowTaskScheduled`
-and `WorkflowTaskStarted`. There might be some events in between if they come after Workflow Task
-was scheduled but not started (e.g., Workflow worker was down and didn't poll for Workflow Task).
-
-**Normal Workflow Task** is created by server every time when server needs Workflow to make progress.
-If Workflow Task fails (worker responds with call to `RespondWorkflowTaskFailed` or an error occurred while
-processing `RespondWorkflowTaskCompleted`) or times out (worker is disconnected),
-server writes corresponding Workflow Task failed event in the history and increase attempt count
-in the mutable state. For the next attempt, Workflow Task events (`WorkflowTaskScheduled`
-and `WorkflowTaskStarted`) are not written into the history but attached to the response of
-`RecordWorkflowTaskStarted` API. These are transient Workflow Task events. Worker is not aware of any "transience"
-of these events. If Workflow Task keeps failing, attempt counter is getting increased in mutable state,
-but no new fail events are written into the history and new transient Workflow Task events are just recreated.
-Workflow Task, which has transient Workflow Task events, is called **transient Workflow Task**.
-When Workflow Task finally completes, `WorkflowTaskScheduled` and `WorkflowTaskStarted` events
-are getting written to history followed by `WorklfowTaskCompleted` event.
+Every Workflow Task ships history events to the worker. It must always contain the two events
+`WorkflowTaskScheduled` and `WorkflowTaskStarted`. There might be some events in-between them if
```

Comment:
I believe it is important to say that WFTStarted must be the last (not just contains).

Reply:
I removed that part because the very next sentence contradicts the original statement that _the last two events_ are WFTScheduled and WFTStarted. I'll change it to emphasize that WFTStart must be last.